Cabal/tests/ParserTests/regressions/version-sets.format

```diff
@@ -0,0 +1,18 @@
+cabal-version: 2.5
+name:          version-sets
+version:       0
+tested-with:
+    ghc ==8.6.3 || ==8.4.4 || ==8.2.2 || ==8.0.2 || ==7.10.3 || ==7.8.4 || ==7.6.3 || ==7.4.2
+synopsis:      version set notation
+
+library
+    default-language: Haskell2010
+    build-depends:
+        network ^>=0,
+        base ==1,
+        base ==1,
+        base ==1 || ==2,
+        base ==1.2,
+        base ==1.2 || ==3.4,
+        ghc ==8.6.3 || ==8.4.4 || ==8.2.2 || ==8.0.2 || ==7.10.3 || ==7.8.4 || ==7.6.3 || ==7.4.2,
```

Comment:
ooh, hmm...

@hvr should we introduce a `VersionRange` constructor for set notation? I'd say yes. And can do it after this is merged.

Reply:
We'd need `NonEmpty` to do it, so it's "a bit" tricky

Reply:
The main benefit would be for syntactical round tripping, wouldn't it? Don't we already have other cases such as `-none` where we have a syntactically lossy encoding?

I wonder if we should instead go for having two different datatypes, one for the concrete parsetree, and one for the desugared abstract syntax tree. The current `VersionRange` type is a bit inconsistent in this regard IMO (i.e. it has support for representing parens which are imo redundant for an expression tree, but otoh it lacks the ability to represent `-none` directly).

In any case, I think we should discuss this in a new separate "design ticket" and we also still have the issue of having to handle pkg-config version-ranges which might be related.